code/controllers/subsystem/plant.dm

```diff
@@ -0,0 +1,89 @@
+var/datum/subsystem/plant/SSplant
+
+// Processes vines/spreading plants.
+/datum/subsystem/plant
+	name          = "Plants"
+	init_order    = SS_INIT_PLANT
+	display_order = SS_DISPLAY_PLANT
+	priority      = SS_PRIORITY_PLANT
+	wait          = 3 SECONDS
+
+	var/list/processing_plants = list()
+	var/list/currentrun
+	var/list/datum/seed/seeds = list() // All seed data stored here.
+	var/list/gene_tag_list = list()    // List of gene tags, how they appear in the centrifuge and bioballistic.
+
+/datum/subsystem/plant/New()
+	NEW_SS_GLOBAL(SSplant)
+
+// Predefined/roundstart varieties use a string key to make it
+// easier to grab the new variety when mutating. Post-roundstart
+// and mutant varieties use their uid converted to a string instead.
+// Looks like shit but it's sort of necessary.
+/datum/subsystem/plant/Initialize()
+	// Populate the global seed datum list.
+	for(var/type in subtypesof(/datum/seed))
+		var/datum/seed/S = new type
+		seeds[S.name] = S
+		S.uid = "[seeds.len]"
+		S.roundstart = TRUE
+	//Might as well mask the gene types while we're at it.
```

Comment:
This block can be replaced by just ´´´gene_tag_list = list(GENE_PHYTOCHEMISTRY, GENE_MORPHOLOGY, GENE_BIOLUMINESCENCE, GENE_ECOLOGY, GENE_ECOPHYSIOLOGY, GENE_METABOLISM, GENE_NUTRITION, GENE_DEVELOPMENT)´´´ since gene masks don't exist anymore